### PR TITLE
First Puppeteer Integration Test

### PR DESCRIPTION
**Pull request by ethersage**

### CODE REVIEW

**Comment by ethersage** [comment]
Lots to consider here. The `manual` app server needs to be started before running the test so I'll have to consider the best way to start them up to run as long as the tests need and for the tests to await the server starting.

Also not completely sold on the test/ folder structure but not worrying about it for now.

File: test/integration/test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const puppeteer = require('puppeteer');
+
+describe('external script', () => {
+    let logs = [];
+
+    beforeAll(async () => {
+        page.on('console', msg => logs.push(msg.text()));
```
Comment (ethersage) [inline test/integration/test.js]: only supports a single page load per test as written

File: test/jest.setup.js [hunk 2]
```diff
@@ -0,0 +1 @@
+jest.setTimeout(100000);
```
Comment (ethersage) [inline test/jest.setup.js]: This is bad ™️. Just for messing around, with fix with a better await timeout pattern

File: test/integration/test.js (hunk 1 above)
Comment (ScriptedAlchemy) [reply test/integration/test.js]: nice log catching

**Comment by ScriptedAlchemy** [comment]
this looks good! we will also need to think of the possible scenarios that the babel compile would fail.

I did this for universal, but pretty much we should snapshot the result of various code transpires. My babel plugin looks at import() and .then() So we wanna make sure that we have a snapshot of a normal promise (like es6 fetch) and a normal import().then() and an import(url).then() to make sure that the code is transpiled right under the circumstances.

File: manual/package.json [hunk 3]
```diff
@@ -37,9 +37,9 @@
     "crc-32": "^1.2.0",
     "cross-env": "^5.2.0",
     "html-webpack-plugin": "^3.2.0",
-    "scriptjs": "^2.5.9",
-    "webpack-merge": "^4.2.1",
     "react": "^16.8.6",
-    "react-dom": "^16.8.6"
+    "react-dom": "^16.8.6",
+    "scriptjs": "^2.5.9",
+    "webpack-merge": "^4.2.1"
```
Comment (ethersage) [inline manual/package.json]: yarn likes to alphabetize things I guess? 🤷‍♂

**Comment by ScriptedAlchemy** [comment]
Perfect!

File: manual/package.json (hunk 3 above)
Comment (ScriptedAlchemy) [reply manual/package.json]: Haha yep! Useful functionality

**Pull request merged**